examples/multigrid-preconditioned-solver/multigrid-preconditioned-solver.cpp

```diff
@@ -125,37 +134,83 @@ int main(int argc, char* argv[])
     // Create smoother factory (ir with bj)
     auto inner_solver_gen =
         gko::share(bj::build().with_max_block_size(1u).on(exec));
+    auto inner_solver_gen_f =
+        gko::share(bj_f::build().with_max_block_size(1u).on(exec));
     auto smoother_gen = gko::share(
         ir::build()
             .with_solver(inner_solver_gen)
-            .with_relaxation_factor(0.9)
+            .with_relaxation_factor(static_cast<ValueType>(0.9))
```

Comment:
These static casts are a bit annoying. Can we remove them here by moving them to where the parameter is being initialized ?

Reply:
Related to #1162 ;)